logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -420,6 +421,18 @@ internal class ConversationDataSource internal constructor(
         }
     }
 
+    override suspend fun fetchSentConnectionConversation(conversationID: ConversationId): Either<CoreFailure, Unit> {
+        return wrapApiRequest {
+            conversationApi.fetchConversationDetails(conversationID.toApi())
+        }.flatMap {
+            val selfUserTeamId = selfTeamIdProvider().getOrNull()
+            val conversation = it.copy(
+                type = ConversationResponse.Type.WAIT_FOR_CONNECTION,
```

Comment:
is this is a hack? I see it's calling the same api as the conversation details.

Reply:
As explained in the ticket here:  https://wearezeta.atlassian.net/browse/WPB-1988?focusedCommentId=115764
In federated environments there will be no change at all, since they can't change it. So in case when the flow it's called to fetch the conversation created by a connection sent, we just "fix" locally the type. Same as other platforms are doing today.

Later after all this federation, mls work, type 3 it's planned to be completely removed from the backend, for all cases, not only federation.

Reply:
Maybe leave TODO here and add proper task that is blocked by backend task?

Reply:
Good idea, done :D